util/ecr.go

```diff
@@ -34,24 +39,24 @@ func ECRCredentialPlugin(plugin map[string]string, pr string) (authConfig types.
 		Region: aws.String(matches[3]),
 	}
 
+	logrus.Debugf("ECRCredentialPlugin: Setting Region to [%s]", matches[3])
 	var sess *session.Session
-	awsAccessKeyID, accessKeyOK := plugin["aws_access_key_id"]
-	awsSecretAccessKey, secretKeyOK := plugin["aws_secret_access_key"]
 
 	// Use predefined keys and override env lookup if keys are present //
-	if accessKeyOK && secretKeyOK {
-		// if session token doesnt exist just pass empty string
-		awsSessionToken := plugin["aws_session_token"]
-		config.Credentials = credentials.NewStaticCredentials(awsAccessKeyID, awsSecretAccessKey, awsSessionToken)
+	if plugin.AwsAccessKeyID != "" && plugin.AwsSecretAccessKey != "" {
+		// if session token doesn't exist just pass empty string
+		config.Credentials = credentials.NewStaticCredentials(plugin.AwsAccessKeyID, plugin.AwsSecretAccessKey, plugin.AwsSessionToken)
 		sess, err = session.NewSession(config)
 	} else {
+		logrus.Debug("ECRCredentialPlugin: aws_access_key_id and aws_secret_access_key keys not in plugin, using IAM role or env variables")
 		sess, err = session.NewSessionWithOptions(session.Options{
 			Config:            *config,
 			SharedConfigState: session.SharedConfigEnable,
 		})
 	}
 
 	if err != nil {
+		logrus.Trace("ECRCredentialPlugin: No error found while constructing auth session, returning authConfig")
```

Comment:
May be a typo here?? `..No error found..` -> `..Error found...`

Reply:
thanks for spotting that. Fixed it up.